### PR TITLE
[FIX] Makes Deathtrap Recycler Movable

### LINKED ISSUE #81205: The dangerous old crusher cannot be unwrenched
Reporting client version: 515.1630

## Issue Summary
The dangerous old crusher cannot be unwrenched, making it impossible to rotate or move it. When you try to rotate it, it says "needs to unwrench!", so it looks like you should be able to unwrench it at least...

The other recyclers (recycler and lumbermill saw) can be unwrenched just fine

## Round ID:
[223120](https://scrubby.melonmesa.com/round/223120)

### PR DESCRIPTION
**Pull request by Higgin**
## About The Pull Request

Fixes #81205 
~~Adds a new flag for letting you move otherwise un-deconstructable objects and making deconstructable objects immovable with a wrench.~~
~~Adds that flag onto the ORM.~~
makes ~~ORM un-unwrenchable,~~ deathtrap recycler unwrenchable.

## Why It's Good For The Game

seems like an oversight on the recycler that unnecessarily limited gimmicks, ~~and one of the long-running peeves with something like the ORM has been how damn easy it is to walk up to it, unwrench it, and walk into cargo through the windoor mapped for mineral storage access (which basically everyone has.) This makes it so if you'd otherwise do that, now you at least need to either deconstruct or destroy the ORM first (or just take another less obvious way in.) Always seemed like an oversight but can atomize it if wanted.~~ atomized out, will ask around before putting back up

## Changelog

:cl:
fix: deathtrap recycler can now be moved.
refactor: moved check for NO_DECONSTRUCTION flag to be inside can_be_unfasten_wrench, allowing us to set specific machines to be movable but not deconstructable.
/:cl:

### CODE REVIEW

File: code/__DEFINES/obj_flags.dm [hunk 1]
```diff
@@ -16,6 +16,7 @@
 #define INFINITE_RESKIN (1<<12) // We can reskin this item infinitely
 #define CONDUCTS_ELECTRICITY (1<<13) //! Can this object conduct electricity?
 #define NO_DECONSTRUCTION (1<<14) //! Prevent deconstruction of this object.
+#define UNWRENCH_BYPASS (1<<15) //! Can this object be unwrenched to move it around?
```
Comment (MrMelbert) [inline code/__DEFINES/obj_flags.dm:19]: I don't think this needs a flag, you can just override `can_be_unfasten_wrench` to return `CANT_UNFASTEN`

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/__DEFINES/obj_flags.dm (hunk 1 above)
Comment (MrMelbert) [reply code/__DEFINES/obj_flags.dm:19]: Hmm actually it might be a bit more complex than I realized, let me think on this.

Comment (Higgin) [reply code/__DEFINES/obj_flags.dm:19]: can_be_unfasten_wrench gets checked inside default_unfasten_wrench after the check for the deconstruction flag, which I'm not sure we want to remove - would it be more appropriate in this case to just snowflake it for the recycler and the ORM?

Comment (MrMelbert) [reply code/__DEFINES/obj_flags.dm:19]: `UNWRENCH_BYPASS` serving double duty is definitely not ideal and I don't like that, I am certain there is a way we can use the procs to our advantage here somewhere but I'm just not sure how

Comment (Higgin) [reply code/__DEFINES/obj_flags.dm:19]: I'll fiddle with it and see how I can do it without the flag.

Comment (Higgin) [reply code/__DEFINES/obj_flags.dm]: you were dead on, works!

**Comment by cnleth** [comment]
The ORM change is a balance change that has nothing to do with the recycler fix. Please separate your changes.

**Comment by DATA-xPUNGED** [comment]
ORM change seems weird, i still wish we got the version where it needs to be unweldeded off the floor first before being unbolted

**Comment by Higgin** [comment]
it really does seem like an oversight, and idk what to tell you if your balance depends on being able to unwrench the ORM, but I'll atomize the ORM change out for now

**Comment by Admiral-Gigaton** [comment]
> > it really does seem like an oversight, and idk what to tell you if your balance depends on being able to unwrench the ORM, but I'll atomize the ORM change out for now
> 
> There were a bunch of PRs related to this (I’m on phone but shouldnt be too hard to find). The problem is if you make it harder to unwrench, tiders will just decon the ORM roundstart so they can get to the lathe for their tools and weapons
> 
> The whole thing is probably more of a mapping issue with cargo being built to be broken into

The solution? 

the all-new NT-Brand (trademark) Assistant zapper auto man 3000! (patented)

now with stingbangs, flashbangs, an auto-harm batong function, flash spammer, pepper spray spammer, and excessive force inducer! it can even burn down the courtroom for you, so you cant get sued! (NT is not liable for brainlet captains calling shuttle for incidents) 

only 12,999,999,999.99 credits! (parts sold separately, batteries not included)

**Pull request merged**